lib/ansible/module_utils/facts/system/os_release.py

```diff
@@ -0,0 +1,51 @@
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+import shlex
+import os
+
+from ansible.module_utils.facts.utils import get_file_content
+
+from ansible.module_utils.facts.collector import BaseFactCollector
+
+
+class OSReleaseFactCollector(BaseFactCollector):
+    name = 'os-release'
+    _fact_ids = set(['os_release'])
+
+    def parse_file(self, path):
+        result = {}
+        for line in get_file_content(path).splitlines():
```

Comment:
I would split this method into:
- reading in the file content to a string
- parsing the string

Then the test cases can test the parsing without having to patch get_file_content or any file io.

May be easier to test if the parsing function was module scope method as well, then the parsing tests dont
need to construct a collector or a faux AnsibleModule. The OsReleaseFactCollector would still be useful for
adding tests for existence of /etc/os-release and general class tests.